plugin/executor/containerd/containerd.go

```diff
@@ -40,7 +54,7 @@ func New(rootDir string, remote libcontainerd.Remote, exitHandler ExitHandler) (
 // Executor is the containerd client implementation of a plugin executor
 type Executor struct {
 	rootDir     string
-	client      libcontainerd.Client
+	client      Client
```

Comment:
Is this mainly so that we can write the unit test with `mockClient`? If so, I think we should continue using `libcontainerd.Client` and think of other ways to rewrite the test.

Reply:
This limits the scope of libcontainerd.Client to just the functions that are called here which makes the mock implementation simpler.

I would indeed like to use the real libcontainerd client implementation, but this is much harder and would either require actually spinning up containerd or building a custom containerd service. The latter sounds nice but is not a simple change.